### PR TITLE
[PVR] Parental check fix; Profile switch fix

### PR DESCRIPTION
**Pull request by ksooo**
Two fixes here:

1) Parental check: Do not show "wrong PIN entered" error box / do not write error log entry if user cancels out parental PIN dialog.

2) Profile switch: Reload PVR manager's settings on profile switch. For example "parentalenabled" value can be different  for different profiles.

Changes are runtime-tested on macOS , latest Kodi master.

@Jalle19 mind taking a look. Changes are straight forward, I guess.

### CODE REVIEW

**Comment by ksooo** [comment]
> TestSharedSection.GetSharedLockWhileTryingExclusiveLock

 ... well known unrelated jenkins error on LINUX-64

**Review by Jalle19 — APPROVED** [review]

**Comment by quthla** [comment]
Can you also fix deadlock when clearing PVR data or epg data via settings?

@ksooo

**Comment by ksooo** [comment]
> Can you also fix deadlock when clearing PVR data or epg data via settings?

 ??? Please don't hijack random PRs to get attention. Use the forum or trac. Thanks.

File: xbmc/pvr/PVRGUIActions.cpp [hunk 1]
```diff
@@ -172,7 +172,7 @@ namespace PVR
   bool CPVRGUIActions::ShowEPGInfo(const CFileItemPtr &item) const
   {
     const CPVRChannelPtr channel(CPVRItem(item).GetChannel());
-    if (channel && !CheckParentalLock(channel))
+    if (channel && CheckParentalLock(channel) != ParentalCheckResult::SUCCESS)
```
[the remark this answers is not in the document]
Comment (ksooo) [reply xbmc/pvr/PVRGUIActions.cpp]: Yes, I am.

screen shot 2018-04-29 at 21 51 56

**Pull request merged**